### PR TITLE
Fix build for OCaml 5

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

File: lru.opam [hunk 1]
```diff
@@ -11,7 +11,7 @@ build: [ [ "dune" "subst" ] {pinned}
          [ "dune" "build" "-p" name "-j" jobs ]
          [ "dune" "runtest" "-p" name ] {with-test & ocaml:version >= "4.07.0"} ]
 depends: [
-  "ocaml" {>="4.03.0"}
+  "ocaml" {>="5.00.0"}
```
Comment (dra27) [inline lru.opam]:
```suggestion
  "ocaml" {>="5.0.0"}
```

**Comment by bikallem** [comment]
Any chance of this PR getting merged? I am getting `ocaml-ci` issues since lru is not compatible with OCaml 5.0 yet. https://github.com/mirage/ocaml-dns/pull/312

**Comment by patricoferris** [comment]
Bumped into this issue recently with Irmin. Trying to install `irmin-graphql.3.4.2` on a `5.0.0` switch can't be done:

```
[ERROR] Package conflict!
  * No agreement on the version of ocaml:
    - (invariant) → ocaml-base-compiler >= 5.0.0~beta1 → ocaml = 5.0.0
    - git-unix >= 3.9.1 → tcpip >= 7.0.0 → lru >= 0.3.0 → ocaml < 5.0
    You can temporarily relax the switch invariant with `--update-invariant'

No solution found, exiting
```

**Comment by pqwy** [comment]
There's no way to make this compatible with pre-5.0, right?

**Comment by edwintorok** [comment]
Why is this using `MakeSeeded` and then providing a seeded hash function that is not seeded?
Wouldn't it be better for `Make` to use `Hashtbl.Make` and `MakeSeeded` to use `Hashtbl.MakeSeeded` and that might even work with both versions?

**Comment by edwintorok** [comment]
I haven't tried, but in principle would this work?
```ocaml
 module FakeSeededHash (H: Hashtbl.HashedType) = struct
    include H 
    let seeded_hash _ x = hash x
    let hash _ x = hash x
  end
  module Make (K: Hashtbl.HashedType) (V: Weighted) =
    Bake (Hashtbl.MakeSeeded (FakeSeededHash (K))) (V)
  module MakeSeeded (K : Hashtbl.SeededHashedType) (V: Weighted) =
    Bake (Hashtbl.MakeSeeded (K)) (V)
```

Then 4.x can pick `hash _ x` from SeededHash module, and 5.x can pick `seeded_hash _ x` and neither would mind having an extra value defined (worst case you get some dead code warning?) that the other version needs.

**Comment by pqwy** [comment]
@edwintorok Right, of course! :)

**Pull request closed without merging**